_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,343 @@
+---
+id: afw4f7
+name: Text node has minimal contrast
+rule_type: atomic
+
+description: |
+  This rule checks that the highest possible contrast of every text node with its background meets the minimal contrast requirement
+
+accessibility_requirements:
+  wcag20:1.4.3: # Contrast (Minimum)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+
+authors:
+  - Brian Bors
+  - Kasper Isager
+  - Wilco Fiers
+---
+
+## Applicability
+
+Any [text node](https://dom.spec.whatwg.org/#text) that is [visible](#visible) and a [child](https://dom.spec.whatwg.org/#concept-tree-child) (in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree)) of an HTML element, except if the [text node](https://dom.spec.whatwg.org/#text) is a [descendant](https://dom.spec.whatwg.org/#concept-shadow-including-descendant) of an element that:
+
+- Has a [semantic role](#semantic-role) that inherits from [widget](https://www.w3.org/TR/wai-aria-1.1/#widget)
+- Is used in the [accessible name](#accessible-name) of a [widget](https://www.w3.org/TR/wai-aria-1.1/#widget) that is [disabled](#disabled) (i.e. It is part of the label)
+- Has a [semantic role](#semantic-role) of [group](https://www.w3.org/TR/wai-aria-1.1/#group) that is [disabled](#disabled)
+
+**Note**: When the text color is the same as the background color, the element is not visible, and so is not applicable.
+
+## Expectation
+
+For each test target, the [highest possible contrast](#highest-possible-contrast) between the [foreground colors](#foreground-colors-of-text) and [background colors](#background-colors-of-text) is at least 4.5:1 or 3.0:1 for [larger scale text](#large-scale-text), except if the [text node](https://dom.spec.whatwg.org/#text) is [decorative](#decorative), or doesn't express anything in [human language](https://www.w3.org/TR/WCAG21/#dfn-human-language-s).
+
+**Note**: Passing this rule does not mean that the [text node](https://dom.spec.whatwg.org/#text) has sufficient color contrast. For example: White text on a black and white image passes this rule as long as it touches at least 1 black pixel. Whether or not that text is legible depends on the how much of the white words are positioned on the black parts of the image. This requires further testing.
```

Comment:
Given how much misunderstanding happened around this rule, even within our small community, I'm in favour of expanding this note to explain the rational behind the choice of the rule.
Namely, that if the highest possible contrast is too small, then it is guaranteed that no contrast is good and the fail-on-fail relationship makes it hard to test further.

Reply:
I've updated the note, please have another read.